server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -236,8 +236,10 @@ private void loadSnapshotInfos(SnapshotsInProgress snapshotsInProgress, String r
         final List<SnapshotsInProgress.Entry> entries = SnapshotsService.currentSnapshots(
             snapshotsInProgress, repositoryName, snapshotIdsToIterate.stream().map(SnapshotId::getName).collect(Collectors.toList()));
         for (SnapshotsInProgress.Entry entry : entries) {
-            snapshotSet.add(new SnapshotInfo(entry));
-            snapshotIdsToIterate.remove(entry.snapshot().getSnapshotId());
+            if (snapshotIdsToIterate.contains(entry.snapshot().getSnapshotId())) {
```

Comment:
We can just do this in one go I think:

```java
if (snapshotIdsToIterate.remove(entry.snapshot().getSnapshotId())) {
    snapshotSet.add(new SnapshotInfo(entry));
}
```

?

Reply:
Yeah, it's better for this situation.